merkle/compact/tree.go

```diff
@@ -94,35 +92,39 @@ func NewTreeWithState(hasher hashers.LogHasher, size int64, getNodesFn GetNodesF
 	for i, id := range ids {
 		r.nodes[id.Level] = hashes[i]
 	}
-	r.recalculateRoot(func(NodeID, []byte) {})
 
-	if !bytes.Equal(r.root, expectedRoot) {
-		glog.Warningf("Corrupt state, expected root %s, got %s", hex.EncodeToString(expectedRoot[:]), hex.EncodeToString(r.root[:]))
-		return nil, fmt.Errorf("root hash mismatch: got %v, expected %v", r.root, expectedRoot)
+	// TODO(pavelkalinnikov): This check should be done externally.
```

Comment:
OTOH having it here forces it to be done...
In the interests of making it hard to hurt yourself with the tools is it work keeping it here?

Reply:
Yeah, it feels safer. On the third hand, maybe there will be use-cases when the expected root hash is unknown, or the check is not necessary for some other reason. I usually advocate for bare minimal interfaces, which is a big pro on its own, and in this case the interface can also be more flexible.